### PR TITLE
doc(agg): add an example with custom UDF spec

### PR DESCRIPTION
**Pull request by shcheklein**
Add an explicit example on an alternative syntax for `agg` (I spent some time to figure our the appropriate signature).

### CODE REVIEW

File: src/datachain/lib/dc.py [hunk 1]
```diff
@@ -904,6 +904,23 @@ def agg(
...
+            # It automatically resolves which columns to pass to the function
+            # by looking at the function signature.
+            def agg_sum(file: list[File],
+                        amount: list[float]) -> Iterator[tuple[File, float]]:
+                yield file[0], sum(amount)
+
+            chain = chain.agg(
+                agg_sum,
+                output={"file": File, "total": float},
+                # Alternative syntax is to use `C`
```
Comment (ilongin) [inline src/datachain/lib/dc.py:918]: I would maybe merge these 2 sentences into one... "Alternative syntax is to use `C` which is useful for nested columns ..."

File: src/datachain/lib/dc.py [hunk 2]
```diff
@@ -904,6 +904,23 @@ def agg(
             )
             chain.save("new_dataset")
             ```
+
+            ```py
+            # It automatically resolves which columns to pass to the function
+            # by looking at the function signature.
+            def agg_sum(file: list[File],
```
Comment (ilongin) [inline src/datachain/lib/dc.py:911]: I would reformat this to be black "compatible"
```python
def agg_sum(
    file: list[File], amount: list[float]
) -> Iterator[tuple[File, float]]:
    yield file[0], sum(amount)
```

**Review by ilongin — APPROVED** [review]
Left 2 small comments

**Comment by shcheklein** [comment]
thanks @ilongin for reviewing it!

**Pull request merged**